CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+cmake_minimum_required (VERSION 3.18)
+
+project(CudaBmk)
+
+set(CMAKE_CUDA_ARCHITECTURES 52 61 70 75)
+enable_language(CUDA)
+
+add_executable(cmem cmem.cu)
+add_executable(diverge diverge.cu diverge2.cu diverge3.cu path.cu)
```

Comment:
The `Makefile-diverge` compiles `diverge2.cu` to `diverge2.cu_sm_12_o` with `-arch sm_12` (because it is listed as `CUFILES_sm_12=diverge2.cu`). Similar arch-specific compiles happen with other targets.

I do not think that behaviour is reproduced here?